src/buf/fixed/pool.rs

```diff
@@ -0,0 +1,353 @@
+        }
+    }
+
+    /// Registers the buffers with the kernel.
+    ///
+    /// This method must be called in the context of a `tokio-uring` runtime.
+    /// The registration persists for the lifetime of the runtime, unless
+    /// revoked by the [`unregister`] method. Dropping the
+    /// `FixedBufPool` instance this method has been called on does not revoke
+    /// the registration or deallocate the buffers.
+    ///
+    /// [`unregister`]: Self::unregister
+    ///
+    /// This call can be blocked in the kernel to complete any operations
+    /// in-flight on the same `io-uring` instance. The application is
+    /// recommended to register buffers before starting any I/O operations.
+    ///
+    /// # Errors
+    ///
+    /// If a collection of buffers is currently registered in the context
+    /// of the `tokio-uring` runtime this call is made in, the function returns
+    /// an error.
+    pub fn register(&self) -> io::Result<()> {
+        crate::io::register_buffers(Rc::clone(&self.inner) as _)
+    }
+
+    /// Unregisters this collection of buffers.
+    ///
+    /// This method must be called in the context of a `tokio-uring` runtime,
+    /// where the buffers should have been previously registered.
+    ///
+    /// This operation invalidates any `FixedBuf` handles checked out from
+    /// this registry instance. Continued use of such handles in I/O
+    /// operations may result in an error.
+    ///
+    /// # Errors
+    ///
+    /// If another collection of buffers is currently registered in the context
+    /// of the `tokio-uring` runtime this call is made in, the function returns
+    /// an error. Calling `unregister` when no `FixedBufPool` is currently
+    /// registered on this runtime also returns an error.
+    pub fn unregister(&self) -> io::Result<()> {
+        crate::io::unregister_buffers(Rc::clone(&self.inner) as _)
+    }
+
+    /// Returns a buffer of requested capacity from this pool
+    /// that is not currently owned by any other [`FixedBuf`] handle.
+    /// If no such free buffer is available, returns `None`.
+    ///
+    /// The buffer is released to be available again once the
+    /// returned `FixedBuf` handle has been dropped. An I/O operation
+    /// using the buffer takes ownership of it and returns it once completed,
+    /// preventing shared use of the buffer while the operation is in flight.
+    ///
+    /// An application should not rely on any particular order
+    /// in which available buffers are retrieved.
+    pub fn try_next(&self, cap: usize) -> Option<FixedBuf> {
+        let mut inner = self.inner.borrow_mut();
+        inner.try_next(cap).map(|data| {
+            let registry = Rc::clone(&self.inner);
+            // Safety: the validity of buffer data is ensured by
+            // Inner::try_next
+            unsafe { FixedBuf::new(registry, data) }
+        })
+    }
+}
+
+// Internal state shared by FixedBufPool and FixedBuf handles.
+pub(crate) struct Inner {
```

Comment:
I just realized I did not make the `Inner` structs module-private, which may cause confusion. This is just a leftover from earlier code. But they are not going to stay private (or named `Inner`) for long, as they will be shared with the upcoming `Sync + Send` fixed buffer implementations.